js/src/personal/haoking/Haoking.page.tsx

```diff
@@ -1,24 +1,50 @@
-import { MantineProvider, Box, Flex, rem } from '@mantine/core'
-import { LeftAboutMe } from './LeftAboutMe.tsx'
-import { RightAboutMe } from './RightAboutMe.tsx'
+import { MantineProvider } from '@mantine/core'
+import { useState } from 'react'
+import { AboutMe } from './aboutme/AboutMe.tsx'
+import { ProjectsPage } from './projects/ProjectsPage.tsx'
+import { MiscPage } from './misc/MiscPage.tsx'
 import { theme } from './theme.ts'
+import { Pages } from './pages.ts'
 
 export function HaokingPage() {
-  return (
-    <MantineProvider theme={theme}>
-      <Box
-        bg={theme.colors.black[4]}
-        h={rem('100vh')}
-        w={rem('100vw')}
-      >
-        <Flex
-          justify="flex-start"
-          direction="row"
-        >
-          <LeftAboutMe />
-          <RightAboutMe />
-        </Flex>
-      </Box>
-    </MantineProvider>
-  )
+  const [current, setCurrent] = useState(Pages.About)
+
+  switch (current) {
```

Comment:
Take a look at AmandaPage to see a version of this without having to have 4 MantineProvider components.

Reply:
fixed on new commit